### ISSUE TITLE
¯\_(ツ)_/¯

### ISSUE THREAD

**Issue by toadkicker** [vkobel/ethdrain#1]
```
325595
325596
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Users/toadkicker/.pyenv/versions/3.6.1/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/toadkicker/.pyenv/versions/3.6.1/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/toadkicker/Projects/ethdrain/ethdrain.py", line 106, in setup_process
    loop.run_until_complete(future)
  File "/Users/toadkicker/.pyenv/versions/3.6.1/lib/python3.6/asyncio/base_events.py", line 466, in run_until_complete
    return future.result()
  File "/Users/toadkicker/Projects/ethdrain/ethdrain.py", line 65, in run
    await asyncio.gather(*tasks)
  File "/Users/toadkicker/Projects/ethdrain/ethdrain.py", line 50, in sema_fetch
    await fetch(url, session, blockNb, fn, actions)
  File "/Users/toadkicker/Projects/ethdrain/ethdrain.py", line 45, in fetch
    if data:
UnboundLocalError: local variable 'data' referenced before assignment
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/toadkicker/Projects/ethdrain/ethdrain.py", line 132, in <module>
    p.map(setup_process, chks)
  File "/Users/toadkicker/.pyenv/versions/3.6.1/lib/python3.6/multiprocessing/pool.py", line 260, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/Users/toadkicker/.pyenv/versions/3.6.1/lib/python3.6/multiprocessing/pool.py", line 608, in get
    raise self._value
UnboundLocalError: local variable 'data' referenced before assignment
```

**Comment by toadkicker** [comment]
<img width="535" alt="screen shot 2017-04-01 at 22 06 51" src="https://cloud.githubusercontent.com/assets/523330/24578979/8c259280-1727-11e7-8bdf-6a6bace1f5a5.png">

**Comment by toadkicker** [comment]
Seems that `data` is in fact called before the blockchain client responds. Probably want [observer patterns](http://code.activestate.com/recipes/131499-observer-pattern/) here?

**Comment by vkobel** [comment]
Fixed with commit 01f72d94a6fea8ed669c5a8256f5fb7fd5a0ef56
